pants-plugins/api_spec/rules.py

```diff
@@ -63,79 +50,39 @@ class APISpecFieldSet(FieldSet):
 
 class GenerateAPISpecViaFmtTargetsRequest(FmtTargetsRequest):
     field_set_type = APISpecFieldSet
-    name = GENERATE_CMD
+    tool_subsystem = GenerateApiSpec
+    partitioner_type = PartitionerType.DEFAULT_SINGLE_PARTITION
 
 
 class ValidateAPISpecRequest(LintTargetsRequest):
     field_set_type = APISpecFieldSet
-    name = VALIDATE_CMD
+    tool_subsystem = ValidateApiSpec
+    partitioner_type = PartitionerType.DEFAULT_SINGLE_PARTITION
 
 
 @rule(
     desc="Update openapi.yaml with st2-generate-api-spec",
     level=LogLevel.DEBUG,
 )
 async def generate_api_spec_via_fmt(
-    request: GenerateAPISpecViaFmtTargetsRequest,
+    request: GenerateAPISpecViaFmtTargetsRequest.Batch,
+    subsystem: GenerateApiSpec,
 ) -> FmtResult:
     # There will only be one target+field_set, but we iterate
     # to satisfy how fmt expects that there could be more than one.
     # If there is more than one, they will all get the same contents.
 
-    # Find all the dependencies of our target
-    transitive_targets = await Get(
-        TransitiveTargets,
-        TransitiveTargetsRequest(
-            [field_set.address for field_set in request.field_sets]
-        ),
-    )
-
-    dependency_files_get = Get(
-        SourceFiles,
-        SourceFilesRequest(
-            sources_fields=[
-                tgt.get(SourcesField) for tgt in transitive_targets.dependencies
-            ],
-            for_sources_types=(FileSourceField, ResourceSourceField),
-        ),
-    )
-
-    source_files_get = Get(
-        SourceFiles,
-        SourceFilesRequest(field_set.source for field_set in request.field_sets),
-    )
+    config_files_get = Get(ConfigFiles, ConfigFilesRequest, subsystem.config_request())
 
     # actually generate it with an external script.
     # Generation cannot be inlined here because it needs to import the st2 code.
```

Comment:
Does this comment make sense now we have removed this chunk of code, and so we no-longer directly call GENERATE_CMD python script here?

Reply:
This code block was moved, not removed. The PexFromTargetsRequest (including all of the random constants used to build it) are now defined on the subsystem. I added a comment to help clarify that.